From 2cc476db69ec75d41d17abbec84ae6472a9e524b Mon Sep 17 00:00:00 2001
From: Mihai Serban <mihai.serban@nxp.com>
Date: Tue, 11 Jul 2017 14:05:40 +0300
Subject: [PATCH 2111/5242] MLK-15951: ARM64: dts: fsl-imx8qxp: Fix SAI master
 clock assignment

commit  fdd4b047459b58100ba933a794e197b11c83b2fa from
https://source.codeaurora.org/external/imx/linux-imx.git

The SAI driver maps the DTS master clock names to the registers settings
that select the audio Master Clock used for internally generated bit clock
when SAI operates in master mode.
The mapping is defined as follows:

mclk0 -> Bus Clock
mclk1 -> Master Clock (MCLK) 1
mclk2 -> Master Clock (MCLK) 2
mclk3 -> Master Clock (MCLK) 3

In iMX8QXP the SAIs are connected to MCLK 1 so we have to use it in DTS.

Signed-off-by: Mihai Serban <mihai.serban@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 02edfd8..421b180 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -1372,8 +1372,8 @@
 		reg = <0x0 0x59040000 0x0 0x10000>;
 		interrupts = <GIC_SPI 314 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&clk IMX8QXP_AUD_SAI_0_IPG>,
-			<&clk IMX8QXP_AUD_SAI_0_MCLK>,
 			<&clk IMX8QXP_CLK_DUMMY>,
+			<&clk IMX8QXP_AUD_SAI_0_MCLK>,
 			<&clk IMX8QXP_CLK_DUMMY>,
 			<&clk IMX8QXP_CLK_DUMMY>;
 		clock-names = "bus", "mclk0", "mclk1", "mclk2", "mclk3";
-- 
1.7.9.5

